service-base/src/main/java/org/eclipse/hono/service/resourcelimits/PrometheusBasedResourceLimitChecks.java

```diff
@@ -470,69 +576,61 @@ private Long extractLongValue(final JsonObject response, final Span span) {
-     *                 {@link TenantConstants#FIELD_MAX_BYTES}. 
-     * @return The allowed data-volume in bytes.
+     * @param configuredLimit The configured limit. 
+     * @return The effective resource limit that has been calculated.
      */
-    long calculateDataVolume(
+    long calculateEffectiveLimit(
             final OffsetDateTime effectiveSince,
             final OffsetDateTime targetDateTime,
             final PeriodMode mode,
-            final long maxBytes) {
+            final long configuredLimit) {
         if (PeriodMode.MONTHLY.equals(mode)
-                && maxBytes > 0
+                && configuredLimit > 0
                 && !targetDateTime.isBefore(effectiveSince)
                 && YearMonth.from(targetDateTime).equals(YearMonth.from(effectiveSince))
                 && effectiveSince.getDayOfMonth() != 1) {
             final OffsetDateTime lastDayOfMonth = effectiveSince.with(TemporalAdjusters.lastDayOfMonth());
             final long daysBetween = ChronoUnit.DAYS
                     .between(effectiveSince, lastDayOfMonth) + 1;
-            return Double.valueOf(Math.ceil(daysBetween * maxBytes / lastDayOfMonth.getDayOfMonth())).longValue();
+            return Double.valueOf(Math.ceil(daysBetween * configuredLimit / lastDayOfMonth.getDayOfMonth()))
+                    .longValue();
         }
-        return maxBytes;
+        return configuredLimit;
     }
 
     /**
-     * Calculates the period for which the data usage is to be retrieved from the
-     * prometheus server based on the mode defined by 
+     * Calculates the period for which the resource usage like volume of used data, connection duration etc. 
+     * is to be retrieved from the Prometheus server based on the mode defined by 
      * {@link TenantConstants#FIELD_PERIOD_MODE}.
      *
-     * @param effectiveSince The point of time on which the data volume limit came
-     *                      into effect.
-     * @param targetDateTime The target point of time used for the data usage period 
-     *                       calculation.
-     * @param mode The mode of the period defined by
-     *                   {@link TenantConstants#FIELD_PERIOD_MODE}.
-     * @param periodInDays The number of days defined by 
-     *                     {@link TenantConstants#FIELD_PERIOD_NO_OF_DAYS}. 
-     * @return The period in days for which the data usage is to be calculated.
+     * @param effectiveSince The point of time on which the resource limit came into effect.
+     * @param currentDateTime The current date and time used for the resource usage period calculation.
+     * @param mode The mode of the period defined by {@link TenantConstants#FIELD_PERIOD_MODE}.
+     * @param periodInDays The number of days defined by {@link TenantConstants#FIELD_PERIOD_NO_OF_DAYS}. 
+     * @return The period in days for which the resource usage is to be calculated.
      */
-    long calculateDataUsagePeriod(
+    long calculateResourceUsagePeriod(
```

Comment:
could this method be moved to `ResourceLimitPeriod`? It doesn't seem to be tied to a Prometheus specific implementation of checks but provides a rather generic helper, doesn't it?

Reply:
I have moved the `calculateResourceUsagePeriod` to `ResourceLimitPeriod`. I have pushed the changes. Would you mind taking a look?